src/core.c/Distribution/Resources.pm6

```diff
@@ -13,26 +13,41 @@ class Distribution::Resources does Associative {
             $!repo-name = Str;
         }
     }
-
     multi method BUILD(:$!dist-id, :$repo, Str :$!repo-name --> Nil) { }
     multi method BUILD(:$!dist-id, Str :$!repo, :$repo-name --> Nil) { }
 
-    method from-precomp() {
-        if %*ENV<RAKUDO_PRECOMP_DIST> -> \dist {
-            my %data := Rakudo::Internals::JSON.from-json: dist;
-            self.new(:repo(%data<repo>), :repo-name(%data<repo-name>), :dist-id(%data<dist-id>))
+    # Alternate instantiator called from Actions.nqp during compilation
+    # of %?RESOURCES
+    method from-precomp(Distribution::Resources:U:) {
```

Comment:
I wonder if this method, which predates `is-implementation-detail`, should be marked as such

Reply:
So done with https://github.com/rakudo/rakudo/commit/b24c5d97dd